src/toil/__init__.py

```diff
@@ -456,214 +452,223 @@ def str_to_datetime(s):
         """
         return datetime.strptime(s, datetime_format)
 
-    class BotoCredentialAdapter(provider.Provider):
-        """
-        Adapter to allow Boto 2 to use AWS credentials obtained via Boto 3's
-        credential finding logic. This allows for automatic role assumption
-        respecting the Boto 3 config files, even when parts of the app still use
-        Boto 2.
+
+from boto import provider
```

Comment:
You can't import from boto at module scope if boto isn't installed. You need something like:

```
try:
    from boto import provider
    base_class = provider.Provider
except ImportError:
    base_class = object

class BotoCredentialAdapter(base_class):
```